doc/tutorials/making_your_first_app.rst

```diff
@@ -164,6 +164,27 @@ add a little more to the plugin.
     :language: python
 
 
+Defining custom settings
+========================
+An important facet of the above plugin is that it takes custom Settings, and has some
+validation built in for those ``Setting`` values. That is, the plugin registers new
+settings that can go in the settings file, and help the user define how the simulation
+runs.
+
+The following example boiler plate code defines three settings. We define two simple
+number settings (inlet and outlet temperatures), and we use :py:class:`Query 
+<armi.operators.settingsValidation .Query>` to define validation on those settings. Here,
+the validation isn't very exciting, we just make sure the temperatures are above zero.
+That's not particularly physically meaningful, but serves as a simple example. The next
+setting is a little more complicated, we define a setting ``myAppVersion`` that defines
+a specific version of our app that this setting file is valid for. And if you try to run
+a different version you get a nasty warning printed to the screen.
```

Comment:
Just so I'm clear, it's now on the plugin / app side to raise an error if they so choose if versions are unsupported?

- Plugin provides a setting for some version
- Plugin defines a setting validator

This works, just want to make sure I understand the paradigm

Reply:
Right now all the `settingsValidation` does is give the option for the user to hand-change a setting interactively. They don't kill a run.  e.g.:

https://github.com/terrapower/armi/blob/e0b01d708eac7ba73c0caabde75158af97fa007e/armi/operators/settingsValidation.py#L51-L61

So, that is the default workflow that I want to show in the `armi-example-app`.

But, in YOUR case, I imagine you would do in the `condition` step of your `Query` is `raise ValueError()` before you `return False`.  That way your run would _actually_ stop if the version was wrong.  Remember, you can put arbitrary code into that first `"condition"` field in a `Query`.

Sorry, I was trying to tow the line between what a general user of the `armi-example-app` would like, and what I thought would be useful to you.

Does that make sense?  Does that seem like a pathway that gets you everything you need/want?

Reply:
Yeah that makes sense and should work. Thanks!

I think also if you use `correction=inspector.NO_ACTION` the `Query` validation will error for you

Reply:
Oh, possibly you are right. I should try that.

How do you feel about this PR?

I know it's not 100% what you wanted, but will it be helpful?  Should we merge it?

Reply:
Oh yes, this is very helpful! The background, context, and implementation all makes a lot more sense.

I'm open to discussing how to improve this later, but I'm also more than content with where this is at